tests/assemble.py

```diff
@@ -16,12 +17,13 @@ def test_parse_line():
     a = Assembler()
     lines = src.splitlines()
     # note: line number = index + 1
-    assert a.parse_line(lines[0]) == None
-    assert a.parse_line(lines[1]) == ('start', 'wait', ('42', ))
-    assert a.parse_line(lines[2]) == (None, 'ld', ('r0', 'r1', '0', ))
-    assert a.parse_line(lines[3]) == (None, 'st', ('r0', 'r1', '0', ))
-    assert a.parse_line(lines[4]) == (None, 'halt', ())
-    assert a.parse_line(lines[5]) == ('end', None, ())
+    assert a.parse_line(lines[0]) == (None, '.set', ('const', '123', ))
+    assert a.parse_line(lines[1]) == None
+    assert a.parse_line(lines[2]) == ('start', 'wait', ('42', ))
+    assert a.parse_line(lines[3]) == (None, 'ld', ('r0', 'r1', '0', ))
+    assert a.parse_line(lines[4]) == (None, 'st', ('r0', 'r1', '0', ))
+    assert a.parse_line(lines[5]) == (None, 'halt', ())
+    assert a.parse_line(lines[6]) == ('end', None, ())
```

Comment:
looks like the testing code (as before your changes) is not diff-friendly, due to these indexes.

maybe using an iterator or just parsing the whole thing would have been simpler.

Reply:
Yip. I intended to keep the code as close to what it was before - which I tend to do to make diff's clean, but of course here it didn't work out. I'm happy to improve this one - still wont be a clean diff, but then for the future it will be easier to make clean changes that.